pkg/logql/engine.go

```diff
@@ -334,7 +334,13 @@ func (q *query) evalSample(ctx context.Context, expr syntax.SampleExpr) (promql_
 	}
 
 	if GetRangeType(q.params) == InstantType {
-		sort.Slice(vec, func(i, j int) bool { return labels.Compare(vec[i].Metric, vec[j].Metric) < 0 })
+		sortByValue, err := Sortable(q.params)
+		if err != nil {
+			return nil, err
+		}
+		if !sortByValue {
+			sort.Slice(vec, func(i, j int) bool { return labels.Compare(vec[i].Metric, vec[j].Metric) < 0 })
```

Comment:
I may be wrong. Just to clarify though. Should this not be 
```
sort.Slice(vec, func(i, j int) bool { return vec[i].V < vec[j].V })
```

Reply:
Because this line of code exists before this PR, the meaning of this line of code seems to be sortByLabel(), but sort and sort_desc need sortByValue(), so when we have sort and sort_desc in our logql grammar, we should not execute sortByLabel ()


sortByValue is here to complete the sorting, and only here can the syntax tree be correctly parsed to accurately locate the need for sort or sort_desc.

```go
} else if expr.Operation == syntax.OpTypeSortDesc {
					result[groupingKey].heap = make(vectorByValueHeap, 0)
					heap.Push(&result[groupingKey].heap, &promql.Sample{
						Point:  promql.Point{V: s.V},
						Metric: s.Metric,
					})
				} else if expr.Operation == syntax.OpTypeSort {
					result[groupingKey].reverseHeap = make(vectorByReverseValueHeap, 0)
					heap.Push(&result[groupingKey].reverseHeap, &promql.Sample{
						Point:  promql.Point{V: s.V},
						Metric: s.Metric,
					})
				}
```

Reply:
Thank you. My bad, I didn't see this code earlier.

Reply:
Thanks for taking the time to review this PR，👍